Experimental/MultiplayerUseCases/Assets/Common/Scripts/NetworkManagerUI.cs

```diff
@@ -1,71 +1,102 @@
-using System;
+using Unity.Netcode.Transports.UTP;
+using Unity.Networking.Transport;
 using UnityEngine;
 using UnityEngine.SceneManagement;
-using UnityEngine.UI;
+using UnityEngine.UIElements;
+using Button = UnityEngine.UIElements.Button;
 
 namespace Unity.Netcode.Samples.MultiplayerUseCases.Common
 {
-
+    [RequireComponent(typeof(UIDocument))]
     public class NetworkManagerUI : MonoBehaviour
     {
-        [SerializeField] Button m_ServerButton;
-        [SerializeField] Button m_HostButton;
-        [SerializeField] Button m_ClientButton;
-        [SerializeField] Button m_DisconnectButton;
-        [SerializeField] Button m_QuitSceneButton;
-        [SerializeField] Color32 m_HighlightedButtonColor;
-        [SerializeField] Color32 m_DisabledButtonColor;
-        [SerializeField] Color32 m_ButtonOutlineColor;
-        [SerializeField] Color32 m_ButtonOutlineHighlightedColor;
-        [SerializeField] Color32 m_ButtonOutlineDisabledColor;
+        const string k_DefaultIP = "127.0.0.1";
+        const string k_DefaultServerListenAddress = "0.0.0.0"; //note: this is not safe for real world usage and would limit you to IPv4-only addresses, but this goes out the scope of this sample.
+        const ushort k_DefaultPort = 7979;
+
+        VisualElement m_Root;
+        VisualElement m_ServerVisualHighlight;
+        VisualElement m_HostVisualHighlight;
+        VisualElement m_ClientVisualHighlight;
+        TextField m_AddressInputField;
+        TextField m_PortInputField;
+        Button m_ServerButton;
+        Button m_HostButton;
+        Button m_ClientButton;
+        Button m_DisconnectButton;
+        Button m_QuitSceneButton;
+
         [SerializeField] string m_SelectionScreenSceneName;
         [SerializeField] GameObject m_ServerOnlyOverlay;
-        Button[] m_Buttons;
 
         void Awake()
         {
-            m_ServerButton.onClick.AddListener(StartServer);
-            m_HostButton.onClick.AddListener(StartHost);
-            m_ClientButton.onClick.AddListener(StartClient);
-            m_DisconnectButton.onClick.AddListener(Disconnect);
-            m_QuitSceneButton.onClick.AddListener(QuitScene);
-            m_Buttons = new Button[] { m_ServerButton, m_HostButton, m_ClientButton };
+            var uiDocument = GetComponent<UIDocument>();
+            m_Root = uiDocument.rootVisualElement;
+
+            m_ServerVisualHighlight = m_Root.Q<VisualElement>("ServerVisualHighlight");
+            m_HostVisualHighlight = m_Root.Q<VisualElement>("HostVisualHighlight");
+            m_ClientVisualHighlight = m_Root.Q<VisualElement>("ClientVisualHighlight");
+            m_AddressInputField = m_Root.Q<TextField>("IPAddressField");
+            m_AddressInputField.SetValueWithoutNotify(k_DefaultIP);
+            m_PortInputField = m_Root.Q<TextField>("PortField");
+            m_PortInputField.SetValueWithoutNotify(k_DefaultPort.ToString());
```

Comment:
You still need to validate inputs here. Please use UIElementsUtils.SetupStringField and provide the appropriate methods.